### DISCUSSION TITLE
Call Vscode Commands from init.lua help

### DISCUSSION THREAD

**Discussion by lowkin18** [vscode-neovim/vscode-neovim#1444] [Q&A] [answered]
I am getting stuck with calling api.call("command") from lua and I can't find any examples

this is what I have in my keymaps.lua

```
if vim.g.vscode then
local neovim_api = require("kipke.core.vscode-neovim")
map("n", "<leader>k", neovim_api.call("editor.action.moveLinesUpAction"), { desc ="create new line and go"}) -- move line up(n)
end
```

I have added the vscode-neovim lua files to the config, I can see this api.call have an affect on startup of vscode as whaterver command is in there runs, but it never fires with  `<leader>k`

Just wondering if someone can explain how i am using this wrong?

**Comment by xiyaowong** [comment]
Not sure what issue you described. Why you copy lua files to your config?

**Comment by justinmk** [comment]
> but it never fires with `<leader>l`

i assume you mean `<leader>k` since that is the mapping you pasted.

**Reply by lowkin18** [reply]
Yes, sorry typo.

These lua files which I assume are needed to make the lua api commands? These have been added into my .config/nvim folder and are used in my init.lua
https://github.com/vscode-neovim/vscode-neovim/tree/master/runtime/lua

This is the .vim setup I assume? What I am asking is how do I replicate this in init.lua with the lua commands? Above is what I have setup for mapping <leader>k to move a line up, but this require.call does nothing. <leader>k is working as expected with other commands. Do I need to append something to this command or am I just using it incorrectly?
<img width="719" alt="image" src="https://github.com/vscode-neovim/vscode-neovim/assets/20023424/abe7f51f-8486-465a-9782-2e12623cfcaa">

I may be out to lunch here as I am relatively new to setting up init.lua with lazy.nvim.

**Comment by lowkin18** [comment]
Am I not supposed to be using the lua variants? 

![image](https://github.com/vscode-neovim/vscode-neovim/assets/20023424/1495b9dc-1152-4adb-af7a-aee45330ad41)

**Reply by xiyaowong** [reply, marked as answer]
Now please use the vim function in Lua by using `fn.function_name` instead. Currently, there is no difference between the two. The Lua API will be redesigned in the future.

**Comment by theol0403** [comment]
You can use Lua commands, but you don't provide a callback/lambda, you just call vscode directly. It tries to run the function on init.
